airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -295,22 +302,6 @@ static boolean isCdc(final JsonNode config) {
     return isCdc;
   }
 
-  /*
-   * It isn't possible to recreate the state of the original database unless we include extra
-   * information (like an oid) when using logical replication. By limiting to Full Refresh when we
-   * don't have a primary key we dodge the problem for now. As a work around a CDC and non-CDC source
-   * could be configured if there's a need to replicate a large non-PK table.
-   *
-   * Note: in place mutation.
-   */
-  private static AirbyteStream removeIncrementalWithoutPk(final AirbyteStream stream) {
-    if (stream.getSourceDefinedPrimaryKey().isEmpty()) {
-      stream.getSupportedSyncModes().remove(SyncMode.INCREMENTAL);
-    }
-
-    return stream;
-  }
```

Comment:
These methods related to CDC catalog manipulation are moved to `PostgresCdcCatalogHelper`.